drivers/at86rf215/at86rf215_o-qpsk.c

```diff
@@ -0,0 +1,276 @@
+{
+    switch (chips) {
+    case BB_FCHIP100:
+    case BB_FCHIP200:
+        return RF_RCUT_FS_BY_5P3;
+    case BB_FCHIP1000:
+        return RF_RCUT_FS_BY_8;
+    case BB_FCHIP2000:
+        return RF_RCUT_FS_BY_4;
+    }
+
+    return 0;
+}
+
+/* Table 6-106. O-QPSK Receiver Frontend Configuration (AGC Settings) */
+static inline uint8_t _AGCC(uint8_t chips)
+{
+    if (chips > BB_FCHIP200) {
+        return (2 << AGCC_AVGS_SHIFT) | AGCC_EN_MASK;
+    } else {
+        return AGCC_EN_MASK;
+    }
+}
+
+/* Table 6-100. MR-O-QPSK Modes */
+static uint32_t _get_bitrate(uint8_t chips, uint8_t mode)
+{
+    switch (chips) {
+    case BB_FCHIP100:
+        return  6250 * (1 << mode);
+    case BB_FCHIP200:
+        return 12500 * (1 << mode);
+    case BB_FCHIP1000:
+    case BB_FCHIP2000:
+        return mode ? 125000 * (1 << (mode - 1)) : 31250;
+    }
+
+    return 0;
+}
+
+static void _set_chips(at86rf215_t *dev, uint8_t chips)
+{
+    /* enable direct modulation if the chip supports it */
+    uint8_t direct_modulation;
+    if (chips < BB_FCHIP1000 && at86rf215_reg_read(dev, RG_RF_VN) >= 3) {
+        direct_modulation = 1 << 4;
+    } else {
+        direct_modulation = 0;
+    }
+
+    /* Set Receiver Bandwidth */
+    at86rf215_reg_write(dev, dev->RF->RG_RXBWC, _RXBWC_BW(chips));
+    /* Set fS; fCUT for RX */
+    at86rf215_reg_write(dev, dev->RF->RG_RXDFE, _RXDFE_SR(chips)
+                                              | _RXDFE_RCUT(chips));
+    /* Set Power Amplifier Ramp Time; fLPCUT */
+    at86rf215_reg_write(dev, dev->RF->RG_TXCUTC, _TXCUTC_PARAMP(chips)
+                                               | _TXCUTC_LPFCUT(chips));
+    /* Set fS; fCUT for TX */
+    at86rf215_reg_write(dev, dev->RF->RG_TXDFE, _TXDFE_SR(chips)
+                                              | _TXDFE_RCUT(chips)
+                                              | direct_modulation);
+
+    /* set receiver gain target according to data sheet */
+    at86rf215_reg_write(dev, dev->RF->RG_AGCS, 3 << AGCS_TGT_SHIFT);
+    at86rf215_reg_write(dev, dev->RF->RG_AGCC, _AGCC(chips));
+
+    /* use RC-0.8 shaping */
+    at86rf215_reg_write(dev, dev->BBC->RG_OQPSKC0, chips | direct_modulation);
+}
+
+static void _set_legacy(at86rf215_t *dev, bool high_rate)
+{
+    uint8_t chips;
+
+    /* enable/disable legacy high data rate */
+    if (high_rate) {
+        at86rf215_reg_write(dev, dev->BBC->RG_OQPSKC3, OQPSKC3_HRLEG_MASK);
+    } else {
+        at86rf215_reg_write(dev, dev->BBC->RG_OQPSKC3, 0);
+    }
+
+    if (is_subGHz(dev)) {
+        chips = BB_FCHIP1000;
+    } else {
+        chips = BB_FCHIP2000;
+    }
+
+    _set_chips(dev, chips);
+
+    at86rf215_reg_write(dev, dev->BBC->RG_OQPSKPHRTX, AT86RF215_OQPSK_MODE_LEGACY);
+
+    at86rf215_reg_write(dev, dev->BBC->RG_OQPSKC2,
+                         RXM_LEGACY_OQPSK       /* receive mode, legacy O-QPSK */
+                       | OQPSKC2_FCSTLEG_MASK   /* 16 bit frame checksum */
```

Comment:
Here and in other places, should it be configurable to 32bit checksum?

Reply:
I thought this was about *removing* features from the driver, not adding new ones 😆

Reply:
hahaha agree, I just though of it when reviewing. postponed.